### DISCUSSION TITLE
Stream your music everywhere using Beetstream (SubSonic API)

### DISCUSSION THREAD

**Discussion by BinaryBrain** [beetbox/beets#4375] [Show and tell]
Hi Beets community,  
I've been working on a Beets plugin called *Beetstream* that allows you to stream your music on SubSonic/AirSonic compatible apps.

What I did was reimplement most of the API endpoints needed by SubSonic clients on a flask server that calls the Beets database.

So if you manage your music library with Beets and want to stream it to your smartphone, you can simply:

1. `pip install beetstream` (https://pypi.org/project/Beetstream)
2. Enable the plugin by writing `plugins: beetstream` in your config file (`~/.config/beets/config.yaml`)
3. Run `beet beetstream` to launch the server
4. Connect to the server using one of the [supported clients](https://github.com/BinaryBrain/Beetstream#supported-clients)

The code is available here: [GitHub - BinaryBrain/Beetstream](https://github.com/BinaryBrain/Beetstream)

Please note that not every endpoint has been implemented. There is still some small features that are missing but overall it works very well. I've been using it with different apps for months now.

Currently all tests has been done manually because many apps crash if the requests are not perfectly formatted. If this plugin gets more popular I'll be happy to code unit tests.

Enjoy listening to your music anywhere!  
Best,  
BinaryBrain

**Comment by arsaboo** [comment]
Just so that I understand, it is a replacement for `navidrome` (or similar programs)?

**Reply by BinaryBrain** [reply]
Yes, exactly! I never used navidrome but the issue I had with this kind of tool is the server often want to handle the music library by itself.
One advantage of SubSonic/AirSonic standard is its wide variety of clients and support.

I wrote some more details on my motivation in the README:
> I personally use Beets to manage my music library on a Raspberry Pi but when I was looking for a way to stream it to my phone I couldn't find any comfortable, suitable and free options.
> I tried [AirSonic](https://airsonic.github.io/) and [SubSonic](http://www.subsonic.org/), [Plex](https://www.plex.tv/) and some other tools but a lot of these solutions want to manage the library as they need (but I prefer Beets) and AirSonic/SubSonic were quite slow and CPU intensive and seemed to have a lot of overhead just to browse albums and send music files. Thus said, SubSonic APIs are good and implemented by a lot of different [clients](https://github.com/BinaryBrain/Beetstream#supported-clients), so I decided to re-implement the server side but based on Beets database (and some piece of code).

**Comment by arsaboo** [comment]
You should also add the plugin [here](https://beets.readthedocs.io/en/stable/plugins/index.html).

**Comment by alienbiker99** [comment]
Thank you for this plugin
